test/unit/test_cloudant_base_error_augment.py

```diff
@@ -0,0 +1,470 @@
+                    # Default to raw body
+                    actual_body = caught_exception.http_response.content
+                    if method != 'HEAD' and mock_content_type:
+                        if mock_content_type.startswith('application/json'):
+                            try:
+                                actual_body = caught_exception.http_response.json()
+                            except JSONDecodeError:
+                                # Catch the malformed JSON case
+                                actual_body = caught_exception.http_response.text
+                        elif mock_content_type == 'text/plain':
+                            actual_body = caught_exception.http_response.text
+                else:
+                    try:
+                        service_response = self._make_request(method, mock_response['stream'])
+                        actual_status_code = service_response.get_status_code()
+                        actual_headers = service_response.get_headers()
+                        actual_body = service_response.get_result()
+                    except Exception as e:
+                        raise Exception('There should be no exception raised in this case.') from e
+                # Assert expected status_code
+                self.assertEqual(mock_response['status'], actual_status_code)
+                # Assert headers unchanged, re-adding the content-type we had to pop for responses mock
+                if mock_content_type:
+                    expected_headers = mock_response['headers'] | {'content-type': mock_content_type}
+                else:
+                    expected_headers = mock_response['headers']
+                self.assertEqual(expected_headers, actual_headers)
+                # Assert expected body
+                self.assertEqual(expected_response['body'], actual_body)
+                return caught_exception
+
+    # Responses mocking doesn't actually go through requests it presents
+    # a requests like API. This means the requests event hooks aren't called
+    # when mocking with responses. However, the response_callback behaves like
+    # requests response event hook so we use it to invoke our _error_response_hook
+    # during testing.
+    @classmethod
+    def _response_callback(cls, resp):
+        return _error_response_hook(resp)
+
+    def test_response_hook_added(self):
+        self.assertEqual(_error_response_hook, self._service.get_http_client().hooks['response'][0])
+
+    def test_response_hook_not_double_added_via_set_http_client(self):
+        client = self._service.get_http_client()
+        self._service.set_http_client(client)
+        response_hooks =  self._service.get_http_client().hooks['response']
+        self.assertEqual(1, len(response_hooks))
+        self.assertEqual(_error_response_hook, response_hooks[0])
+
+    def test_response_hook_added_via_set_http_client(self):
```

Comment:
This test seem to me a duplication of `test_response_hook_not_double_added_via_set_http_client`.

Reply:
right, I missed that line!